core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/lru/CachedBlockQueue.java

```diff
@@ -91,24 +93,19 @@ public void add(CachedBlock cb) {
    * @return list of cached elements in descending order
    */
   public CachedBlock[] get() {
-    LinkedList<CachedBlock> blocks = new LinkedList<>();
-    while (!queue.isEmpty()) {
-      blocks.addFirst(queue.poll());
-    }
-    return blocks.toArray(new CachedBlock[blocks.size()]);
+     CachedBlock[] blocks = queue.toArray(new CachedBlock[0]);
```

Comment:
Thanks for the review @ctubbsii ! This is the correct way to call `toArray()`.

Check out the example here:

https://docs.oracle.com/javase/8/docs/api/?java/util/Collections.html

>Returns an array containing all of the elements in this collection; the runtime type of the returned array is that of the specified array. If the collection fits in the specified array, it is returned therein. Otherwise, *a new array is allocated* with the runtime type of the specified array and the size of this collection.

The `new CachedBlock[0]` is a no-op since an array of size zero cannot be resized and cannot contain any elements, therefore, it is completely immutable.  The compiler will just create a single global instance of the empty array.  It does not create a new empty array for each invocation at runtime.

Reply:
Calling with an empty array may be optimized out by the compiler, but that behavior isn't guaranteed, and I still think it makes for a strange style to pass in an array that you know in advance that you will not be using at all. But, I'm sure this is just personal preference / style choice, so it's fine.